Lib/fontTools/misc/psCharStrings.py

```diff
@@ -1102,6 +1102,15 @@ def toXML(self, xmlWriter):
 					args = []
 				else:
 					args.append(token)
+			if args:
```

Comment:
I find a bit odd that a method from a super-class checks if `self` is an instance of one of the sub-classes..
Maybe the `if args: ...` branch could be refactored into a method `_checkLastStackItems` or a similar, and the `T2Charstring` default implementation would assert, whereas the `CFF2Subr` subclass would override it and write the extra arguments. Or maybe it's ok.

Reply:
Yes, this was inelegant. I thought a bit about minimum call overhead, and just added a class field 'isCFF2' that is False for T2Charstring and True for CFFCharstring. I chose 'isCFF2', as this is used a number of other class. If you prefer another, let me know, or just change it directly - I have no attachment to any particular solution.